auth/sasl/mech/crammd5/mechanism.go

```diff
@@ -0,0 +1,67 @@
+package crammd5
+
+import (
+	"crypto/hmac"
+	"crypto/md5"
+	"encoding/hex"
+	"errors"
+	"io"
+
+	log "github.com/golang/glog"
+	"github.com/mesos/mesos-go/auth/sasl/callback"
+	"github.com/mesos/mesos-go/auth/sasl/mech"
+)
+
+var (
+	illegalStateErr = errors.New("illegal mechanism state")
```

Comment:
Should this be moved to the `mech` package and exported (along with `illegalState` step function)?